src/cunumeric/index/zip.cc

```diff
@@ -34,6 +34,7 @@ struct ZipImplBody<VariantKind::CPU, DIM, N> {
                   bool dense,
                   const int64_t key_dim,
                   const int64_t start_index,
+                  const DomainPoint shape,
```

Comment:
I think it's better to pass this by reference, as a `const DomainPoint&` (here and in `*.cu`, `*_omp.cc`).